File: leetcode/026_removeduplicatesfromsortedarray.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if len(nums) <= 1:
            return len(nums)
        r = 1
        w = 1
        while r < len(nums):
            if nums[r] != nums[r-1]:
                nums[w] = nums[r]
                w += 1
            r += 1
        return w

File: leetcode/026_removeduplicatesfromsortedarray.ts
function removeDuplicates(nums: number[]): number {
    let writeIdx = 1
    let readIdx = 1
    
    while (readIdx < nums.length) {
        if (nums[readIdx] === nums[writeIdx - 1]){
            readIdx = readIdx + 1
            continue
        } 

        nums[writeIdx] = nums[readIdx]
        writeIdx = writeIdx + 1
        readIdx = readIdx + 1
    }

    return writeIdx
};
